### ISSUE TITLE
sumologic_caller_identity is questionable

### ISSUE THREAD

**Issue by jolexa** [SumoLogic/sumologic-terraform-provider#34]
Hello,

I find the ability to fetch the access key pretty dangerous. Can someone please describe the reasoning for this ability? If we want to avoid leaking the key into terraform state, this ability should not be available.

**Comment by jolexa** [comment]
Hi @bendrucker ,

`aws_caller_identity` is great for templating ARNs and other use-cases. I am a heavy user of that data source and understand it well.

Can you describe a scenario where being able to use a data source to fetch the sumo credentials (and then store them in state) is useful? I'm sure there are scenarios, but I am struggling to think of them.

On the contrary, the security risk of storing credentials in state is real. Our organization uses [Atlantis](https://www.runatlantis.io/) to power our TF workflow and store remote state in S3. We have the Sumo access keys available to Atlantis so that anyone in our organization can deploy a Sumo Collector, this works well and can be audited. However, once credentials are exposed to end consumers, we start losing the ability to audit and the possibility of leaking credentials becomes much higher.

**Comment by horjulf** [comment]
I agree, it shouldn't return the `access_key` and only expose the called id `access_id`.

**Comment by frankreno** [comment]
Unless anyone else wildly disagrees, I will go with @horjulf's suggestion and remove `access_key` but keep `access_id`.
